src/js/popup/popup.js

```diff
@@ -1,802 +1,1350 @@
+          return await browser.runtime.sendMessage({
+            method: "getServerStoragePref",
+          });
+        } else {
+          // If the stored pref exists, return value
+          return serverStoragePref.server_storage;
+        }
+      },
+      getCurrentPage: async () => {
+        const [currentTab] = await browser.tabs.query({
+          active: true,
+          currentWindow: true,
+        });
+        return currentTab;
+      },
+      getMasks: async (options = { fetchCustomMasks: false }) => {
+        const serverStoragePref =
+          await popup.utilities.getCachedServerStoragePref();
+
+        if (serverStoragePref) {
+          try {
+            return await browser.runtime.sendMessage({
+              method: "getAliasesFromServer",
+              options,
+            });
+          } catch (error) {
+            console.warn(`getAliasesFromServer Error: ${error}`);
+
+            // API Error — Fallback to local storage
+            const { relayAddresses } = await browser.storage.local.get(
+              "relayAddresses"
+            );
+
+            return relayAddresses;
+          }
+        }
+
+        // User is not syncing with the server. Use local storage.
+        const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+        return relayAddresses;
+      },
+      populateNewsFeed: async ()=> {
+        // audience can be premium, free, phones, all
+        // Optional data: waffle, fullCta*
+        const savings = "40%"; // For "Save 40%!" in the Bundle promo body
+        
+        const isBundleAvailableInCountry = (await browser.storage.local.get("bundlePlans")).bundlePlans.BUNDLE_PLANS.available_in_country;
+        const isPhoneAvailableInCountry = (await browser.storage.local.get("phonePlans")).phonePlans.PHONE_PLANS.available_in_country;
+        const hasPhone = (await browser.storage.local.get("has_phone")).has_phone;
+        const hasVpn = (await browser.storage.local.get("has_vpn")).has_vpn;
+
+        // Conditions for phone masking announcement to be shown: if the user is in US/CAN, phone flag is on, and user has not purchased phone plan yet
+        const isPhoneMaskingAvailable = isPhoneAvailableInCountry && !hasPhone;
+
+        // Conditions for bundle announcement to be shown: if the user is in US/CAN, bundle flag is on, and user has not purchased bundle plan yet
+        const isBundleAvailable = isBundleAvailableInCountry && !hasVpn;
+      
+        // Conditions for firefox integration to be shown: if the waffle flag "firefox_integration" is set as true
+        const isFirefoxIntegrationAvailable = await checkWaffleFlag("firefox_integration");
+        
+        // FIXME: The order is not being set correctly
+        if (isFirefoxIntegrationAvailable) {
+          state.newsContent.push({
+            id: "firefox-integration",
+            waffle: "firefox_integration",
```

Comment:
Is this not redundant with the `checkWaffleFlag` call above on line 1245?

And in general, when should we use the conditional function and when one of the properties specific to particular conditions? I imagine it might be easier to read to have all the conditions for a particular news item in one place?

Reply:
Yeah. See above comment from @codemist. I removed the `logicCheck` field and logic in ff693f4b83cc29639a606c1cfacb37619a0f61c8